From 16834df3c365234f0b3f201cf38aaf9f1860bf81 Mon Sep 17 00:00:00 2001
From: spena <spena@unknown>
Date: Fri, 10 Apr 2015 17:10:45 +0000
Subject: [PATCH 0112/1164] HIVE-10202 Beeline outputs prompt+query on
 standard output when used in non-interactive mode
 (Naveen Gangam via Sergio Pena)

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1672702 13f79535-47bb-0310-9956-ffa450edef68
---
 .../src/java/org/apache/hive/beeline/Commands.java |   18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/Commands.java b/beeline/src/java/org/apache/hive/beeline/Commands.java
index 6c3cdb5..4f9aae2 100644
--- a/beeline/src/java/org/apache/hive/beeline/Commands.java
+++ b/beeline/src/java/org/apache/hive/beeline/Commands.java
@@ -716,14 +716,22 @@ private boolean execute(String line, boolean call) {
       while (beeLine.getConsoleReader() != null && !(line.trim().endsWith(";"))
         && beeLine.getOpts().isAllowMultiLineCommand()) {
 
-        StringBuilder prompt = new StringBuilder(beeLine.getPrompt());
-        for (int i = 0; i < prompt.length() - 1; i++) {
-          if (prompt.charAt(i) != '>') {
-            prompt.setCharAt(i, i % 2 == 0 ? '.' : ' ');
+        if (!beeLine.getOpts().isSilent()) {
+          StringBuilder prompt = new StringBuilder(beeLine.getPrompt());
+          for (int i = 0; i < prompt.length() - 1; i++) {
+            if (prompt.charAt(i) != '>') {
+              prompt.setCharAt(i, i % 2 == 0 ? '.' : ' ');
+            }
           }
         }
 
-        String extra = beeLine.getConsoleReader().readLine(prompt.toString());
+        String extra = null;
+        if (beeLine.getOpts().isSilent() && beeLine.getOpts().getScriptFile() != null) {
+          extra = beeLine.getConsoleReader().readLine(null, jline.console.ConsoleReader.NULL_MASK);
+        } else {
+          extra = beeLine.getConsoleReader().readLine(beeLine.getPrompt());
+        }
+
         if (extra == null) { //it happens when using -f and the line of cmds does not end with ;
           break;
         }
-- 
1.7.9.5

